src/osqp.c

```diff
@@ -232,18 +230,18 @@ c_int osqp_setup(OSQPWorkspace** workp,
   work->pol->z         = OSQPVectorf_malloc(m);
   work->pol->y         = OSQPVectorf_malloc(m);
   if (!(work->pol->x)) return osqp_error(OSQP_MEM_ALLOC_ERROR);
-  if ((!(work->pol->Alow_to_A) || !(work->pol->Aupp_to_A) ||
+  if (!(work->pol->Alow_to_A) || !(work->pol->Aupp_to_A) ||
       !(work->pol->A_to_Alow) || !(work->pol->A_to_Aupp) ||
-      !(work->pol->z) || !(work->pol->y)) )
+      !(work->pol->z) || !(work->pol->y))
     return osqp_error(OSQP_MEM_ALLOC_ERROR);
 
   // Allocate solution
   work->solution = c_calloc(1, sizeof(OSQPSolution));
   if (!(work->solution)) return osqp_error(OSQP_MEM_ALLOC_ERROR);
   work->solution->x = c_calloc(1, n * sizeof(c_float));
   work->solution->y = c_calloc(1, m * sizeof(c_float));
-  if (!(work->solution->x))            return osqp_error(OSQP_MEM_ALLOC_ERROR);
-  if (!(work->solution->y)) return osqp_error(OSQP_MEM_ALLOC_ERROR);
+  if (!(work->solution->x))      return osqp_error(OSQP_MEM_ALLOC_ERROR);
+  if (m && !(work->solution->y)) return osqp_error(OSQP_MEM_ALLOC_ERROR);
```

Comment:
`mxMalloc` returns `NULL` when allocating zero memory, which was causing an error when solving an unconstrained problem (`m=0`).

Reply:
I thought this was what always happened, but apparently not.   Would maybe be better to explicitly set the pointer to NULL instead of allocating in the first place if m = NULL.   I think I did this in the OSQPVectorx types, but for the solution struct I guess you are working with bare arrays still.

Reply:
Yes, there were no problems with `OSQPVectorx`, only with bare arrays.